plugins/amdgpu/amdgpu_plugin.c

```diff
@@ -1036,11 +1041,81 @@ static int dump_bos(int fd, struct kfd_ioctl_criu_process_info_args *info_args,
 	return ret;
 }
 
+static int dump_queues(int fd, struct kfd_ioctl_criu_process_info_args *info_args, CriuKfd *e)
+{
+	struct kfd_ioctl_criu_dumper_args args = {0};
+	struct kfd_criu_queue_bucket *queue_buckets;
+	uint8_t *priv_data;
+	int ret = 0, i;
+
+	pr_debug("Dumping %d queues\n", info_args->total_queues);
+
+	if (!info_args->total_queues)
+		return 0;
+
+	args.type = KFD_CRIU_OBJECT_TYPE_QUEUE;
+	args.objects_index_start = 0; /* Partial object lists not supported for now */
+
+	args.num_objects = info_args->total_queues;
+	args.objects_size = (args.num_objects * sizeof(*queue_buckets)) +
+				info_args->queues_priv_data_size;
+
+	args.objects = (uintptr_t)xmalloc(args.objects_size);
+	if (!args.objects) {
+		ret = -ENOMEM;
+		goto exit;
+	}
+
+	ret = kmtIoctl(fd, AMDKFD_IOC_CRIU_DUMPER, &args);
+	if (ret) {
+		pr_perror("amdgpu_plugin: Failed to call dumper (queues) ioctl");
+		goto exit;
+	}
+
+	queue_buckets = (struct kfd_criu_queue_bucket*)args.objects;
+	/* First private data starts after all buckets */
+	priv_data = (uint8_t *)args.objects + (args.num_objects * sizeof(*queue_buckets));
+
+	e->num_of_queues = info_args->total_queues;
+	ret = allocate_q_entries(e, args.num_objects);
```

Comment:
Use e->num_of_queues instead of args.num_objects.

Reply:
ACK